pkg/mcs/resource_manager/client/client.go

```diff
@@ -503,9 +502,11 @@ func (gc *groupCostController) handleTokenBucketTrickEvent(ctx context.Context)
 				counter.setupNotificationCh = nil
 				counter.limiter.SetupNotificationThreshold(gc.run.now, counter.setupNotificationThreshold)
 				gc.updateRunState(ctx)
-			default:
+			case <-ctx.Done():
+				return
 			}
 		}
+
 	case rmpb.GroupMode_RUMode:
 		for _, counter := range gc.run.requestUnitTokens {
 			select {
```

Comment:
Should check whether counter.setupNotificationCh is nil. Otherwize, if there are multi counter, it can cause deadlock.

Reply:
good catch, fixed.